### PR TITLE
Sync upstream v1.30.1

### LINKED ISSUE #304: Issue for - Sync with Upstream v1.30
## What would you like to be added
Gardener autoscaler should be synced with respective [Kubernetes Autoscaler 1.30.0 Release](https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.30.0)

## Why is this needed

* To keep the fork in sync with upstream.
* To ensure the shoot control planes are using respective version of CA for a given K8S version

## Steps

Please follow the steps in
[How do I sync gardener autoscaler with an upstream autoscaler minor release](https://github.com/gardener/autoscaler/blob/machine-controller-manager-provider/cluster-autoscaler/FAQ.md#how-do-i-sync-gardener-autoscaler-with-an-upstream-autoscaler-minor-release)

### PR DESCRIPTION
**Pull request by sssash18**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #304 

**Special notes for your reviewer**:

**Release note**:

```other operator
- Synced changes till v1.30.1 of upstream autoscaler.
```

### CODE REVIEW

File: cluster-autoscaler/integration/logs/autoscaler_processs.log [hunk 1]
```diff
@@ -0,0 +1,4061 @@
+I0708 14:05:09.210791   60673 main.go:647] Cluster Autoscaler 1.30.1
```
Comment (aaronfern) [inline cluster-autoscaler/integration/logs/autoscaler_processs.log]: This file is added by error I guess. Can you please remove it?

Comment (sssash18) [reply cluster-autoscaler/integration/logs/autoscaler_processs.log]: done

File: cluster-autoscaler/SYNC-CHANGES/SYNC_CHANGES-1.30.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+- [v1.30.1](#v1290)
+    - [Synced with which upstream CA](#synced-with-which-upstream-ca)
+    - [Changes made](#changes-made)
+        - [During merging](#during-merging)
+        - [During vendoring k8s](#during-vendoring-k8s)
+        - [Others](#others)
+
+
+# v1.29.0
```
Comment (aaronfern) [inline cluster-autoscaler/SYNC-CHANGES/SYNC_CHANGES-1.30.md]:
```suggestion
# v1.30.0
```

File: cluster-autoscaler/SYNC-CHANGES/SYNC_CHANGES-1.30.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+
+## Synced with which upstream CA
+
+[v1.30.1](https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.30.1)
+
+## Changes made
```
Comment (aaronfern) [inline cluster-autoscaler/SYNC-CHANGES/SYNC_CHANGES-1.30.md:18]: There are 2 flags added as part of 1.30 `scale-down-delay-type-local` and `node-group-keep-backoff-out-of-resources`. Please add these to the SYNC notes

File: cluster-autoscaler/core/scaledown/eligibility/eligibility.go [hunk 4]
```diff
@@ -164,9 +164,7 @@ func (c *Checker) unremovableReasonAndNodeUtilization(context *context.Autoscali
 	}
 
 	// FORK-CHANGE: log added to identify underutilized nodes
-	klog.V(2).Infof("Node %s is underutilized: %s requested (%.6g%% of allocatable) is below the scale-down utilization threshold", node.Name, utilInfo.ResourceName, utilInfo.Utilization*100)
-
-	klogx.V(4).UpTo(utilLogsQuota).Infof("Node %s - %s requested is %.6g%% of allocatable", node.Name, utilInfo.ResourceName, utilInfo.Utilization*100)
+	klogx.V(2).UpTo(utilLogsQuota).Infof("Node %s - %s requested is %.6g%% of allocatable", node.Name, utilInfo.ResourceName, utilInfo.Utilization*100)
```
Comment (aaronfern) [inline cluster-autoscaler/core/scaledown/eligibility/eligibility.go:168]:
```suggestion
    klog.V(2).Infof("Node %s is underutilized: %s requested (%.6g%% of allocatable) is below the scale-down utilization threshold", node.Name, utilInfo.ResourceName, utilInfo.Utilization*100)
	klogx.V(2).UpTo(utilLogsQuota).Infof("Node %s - %s requested is %.6g%% of allocatable", node.Name, utilInfo.ResourceName, utilInfo.Utilization*100)
```

File: cluster-autoscaler/.gitignore [hunk 5]
```diff
@@ -13,4 +11,3 @@ main
 Session.vim
 .netrwhist
 .vscode
-/integration/logs
```
Comment (aaronfern) [inline cluster-autoscaler/.gitignore:16]: Please revert this line

**Review by aaronfern — CHANGES_REQUESTED** [review]
Thanks for the PR!
Few minor changes requested. Please address them

**Review by aaronfern — CHANGES_REQUESTED** [review]
Thanks for the PR!
Few minor changes requested. Please address them

File: cluster-autoscaler/SYNC-CHANGES/SYNC_CHANGES-1.30.md (hunk 3 above)
Comment (aaronfern) [reply cluster-autoscaler/SYNC-CHANGES/SYNC_CHANGES-1.30.md:18]: Resolving after an offline discussion 
1. `scale-down-delay-type-local` has been included from upstream `1.29.2` and is already part of our `1.29.0` sync changes
2. `node-group-keep-backoff-out-of-resources` even though it is mentioned in the release notes, it is not present in the codebase.

**Review by aaronfern — APPROVED** [review]
Thanks for the PR!
/lgtm

**Pull request merged**